projects/optic/src/commands/oas/verify.ts

```diff
@@ -188,6 +221,62 @@ export async function verifyCommand(): Promise<Command> {
         },
       });
 
+      if (options.upload) {
+        if (!opticUrlDetails) {
+          console.error(
+            `File ${specPath} does not have an optic url. Files must be added to Optic and have an x-optic-url key before verification data can be uploaded.`
+          );
+          console.error(
+            `${chalk.yellow('Hint: ')} Run optic api add ${specPath}`
+          );
+          process.exitCode = 1;
+          return;
+        }
+
+        const { orgId, apiId } = opticUrlDetails;
+        const tags: string[] = [];
+        if (config.vcs?.type === VCS.Git) {
+          tags.push(`git:${config.vcs.sha}`);
+          const currentBranch = await Git.getCurrentBranchName();
+          tags.push(sanitizeGitTag(`gitbranch:${currentBranch}`));
+        }
+        const specId = await uploadSpec(apiId, {
+          spec: parseResult,
+          client: config.client,
+          tags,
+          orgId,
+        });
+
+        const endpointHashes = {};
+        for (const [path, pathObj] of Object.entries(
+          parseResult.jsonLike.paths
+        )) {
+          if (pathObj) {
+            for (const [method, methodObj] of Object.entries(pathObj)) {
+              if (
+                Object.values(OpenAPIV3.HttpMethods).includes(method as any)
+              ) {
+                const endpointChecksum = computeEndpointChecksum(
+                  path,
+                  method,
+                  methodObj as OpenAPIV3.OperationObject
+                );
+                const pathHashes = endpointHashes[path] ?? {};
+                pathHashes[method] = endpointChecksum;
+                endpointHashes[path] = pathHashes;
```

Comment:
I'm computing the endpoint hashes separately, outside of the coverage datastructure. Keying this with the non-denormalized path